File: kits/kotlin/simple/bot/src/main/kotlin/luxaibot/lux/GameMap.kt
package luxaibot.lux

class GameMap(val width: Int, val height: Int) {
    val map: Array<Array<Cell?>>

    init {
        map = Array<Array<Cell?>>(height) { arrayOfNulls<Cell>(width) }
        for (y in 0 until height) {
            for (x in 0 until width) {
                map[y][x] = Cell(x, y)
            }
        }
    }

    fun getCellByPos(pos: Position): Cell? {
        return map[pos.y][pos.x]
    }

    fun getCell(x: Int, y: Int): Cell? {
        return map[y][x]
    }

    /** Internal use only  */
    internal fun _setResource(rType: String, x: Int, y: Int, amount: Int) {
        val cell: Cell? = getCell(x, y)
        cell!!.resource = Resource(rType, amount)
    }
}

File: kits/csharp/simple/Bot/Lux/GameMap.cs
using System;

namespace Bot.Lux
{
	public class GameMap
	{
		public int Width;
		public int Height;
		public Cell[][] Map;

		public GameMap(int width, int height)
		{
			Width = width; 
			Height = height;
			Map = new Cell[height][];

			for (var y = 0; y < Height; y++)
			{
				Map[y] ??= new Cell[width];
				for (var x = 0; x < Width; x++)
				{
					Map[y][x] = new Cell(x, y);
				}
			}
		}

		public Cell GetCellByPos(Position pos)
		{
			return Map[pos.Y][pos.X];
		}

		public Cell GetCell(int x, int y)
		{
			return Map[y][x];
		}

		/** Internal use only */
		public void SetResource(String rType, int x, int y, int amount)
		{
			var cell = GetCell(x, y);
			cell.Resource = new Resource(rType, amount);
		}
	}
}
